### ISSUE TITLE
Training segmentation model on COCO dataset with panoptic annotations - performance issues

### ISSUE THREAD

**Issue by xperroni** [ultralytics/ultralytics#20332]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I wrote a [script](https://github.com/xperroni/annotate_for_yolo) to convert the 133-class COCO panoptic segmentation annotations to the [YOLO instance segmentation dataset format](https://docs.ultralytics.com/datasets/segment/). I then used the converted annotations to train the `yolo11n-seg` model with:

    yolo segment train data=coco_panoptic.yaml model=yolo11n-seg.pt epochs=600

However, training results were rather underwhelming. By the 300th epoch mAP(box)50-95 has all but converged to around 22.0, and mAP(mask)50-95 to around 18.0. These are way below the mAP(box)50-95 of 38.9 and the mAP(mask)50-95 of 32.0 [reported](https://docs.ultralytics.com/tasks/segment/#models) for the `yolo11n-seg` model trained on COCO with Ultralytics' custom 80-class segmentation annotations.

How do I improve those numbers? I know there are various [training settings](https://docs.ultralytics.com/modes/train/#resuming-interrupted-trainings) I could play with, but it's hard to know where to start. Is there anywhere I can find the settings used for the Ultralytics pretrained models?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xperroni, thank you for your detailed question and for using Ultralytics 🚀! This is an automated response to help you get started while an Ultralytics engineer reviews your issue and will assist you further soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for insights into [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as answers to many common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven't already. This helps our team debug more effectively.

If you have a custom training ❓ Question (like this one), please provide as much detail as possible, including sample images from your dataset, training logs, and confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For real-time chat and support, join our [Discord](https://discord.com/invite/ultralytics) 🎧. If you prefer more in-depth discussion, visit [Discourse](https://community.ultralytics.com/), or join conversations on our [Subreddit](https://reddit.com/r/Ultralytics) to connect with the community.

## Upgrade

Ensure you’re using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for reaching out and sharing your experience! An Ultralytics engineer will follow up soon to help you further.

**Comment by Y-T-G** [comment]
Can you post the plots of the label visualizations generated inside runs folder?

**Comment by xperroni** [comment]
Sorry for the late response. Do you mean these?

![Image](https://github.com/user-attachments/assets/8860b1b3-8a85-4c1d-afdc-de507dd6c8a6)
![Image](https://github.com/user-attachments/assets/a2432641-5322-4ae2-a8cb-0da7891b2629)
![Image](https://github.com/user-attachments/assets/4e604d53-29d6-43a4-b4d4-f9d4def6c714)
![Image](https://github.com/user-attachments/assets/8796a98c-4c82-41ba-866b-65c43d683f76)
![Image](https://github.com/user-attachments/assets/71eaf09c-04d0-444b-a22a-e42e64acb80c)

**Comment by glenn-jocher** [comment]
Looking at your label visualizations, I can see a potential issue. The distribution across 133 classes appears quite imbalanced compared to the standard 80-class COCO dataset used in the pretrained model. This difference in class structure could significantly impact performance.

A few suggestions to improve your results:

1. Try using a larger model like YOLO11s-seg or YOLO11m-seg which have more capacity to handle the increased number of classes

2. Adjust your training hyperparameters - particularly increase the learning rate warmup epochs and potentially reduce the overall learning rate

3. Check your annotation quality - the conversion from panoptic to instance segmentation might be introducing artifacts or inconsistencies

4. Consider implementing class weights or focal loss to address class imbalance issues

The pretrained models are typically trained with extensive augmentation, optimized hyperparameters, and carefully curated datasets. You might want to examine the default hyperparameters in the Ultralytics codebase as a starting point, particularly in the configuration files under the cfg directory.

**Comment by xperroni** [comment]
> Consider implementing class weights or focal loss to address class imbalance issues

Doesn't the Ultralytics package already implement focal loss? I thought that's what the `dfl_loss` column in the training script output was about.

> You might want to examine the default hyperparameters in the Ultralytics codebase as a starting point, particularly in the configuration files under the cfg directory.

I looked into the `cfg/` directory but couldn't find anything in terms of what hyperparameters were used to train the pretrained models. Am I missing something?

**Comment by glenn-jocher** [comment]
The `dfl_loss` in the training output actually refers to "Distribution Focal Loss" which is specifically for bounding box regression, not the classification focal loss you're thinking of. For classification, Ultralytics does implement a variant of focal loss by default, but this is separate from what you're seeing in the output.

Regarding hyperparameters for pretrained models, you won't find them explicitly documented in the `cfg/` directory. The default training hyperparameters are defined in `ultralytics/cfg/default.yaml` which serves as a starting point. For segmentation tasks specifically, check the model-specific configurations under `ultralytics/cfg/models/11` especially the `-seg` variants.

For your 133-class problem, I'd recommend:

1. Using `imgsz=640` or larger to capture more detail
2. Extending training duration to 1000+ epochs 
3. Implementing a learning rate scheduler with a longer warmup
4. Increasing model size as previously suggested

Try examining a few successful training runs with `yolo segment train model=yolo11s-seg.pt data=coco8-seg.yaml` first to see the default hyperparameters in action before scaling up to your full dataset.

**Comment by xperroni** [comment]
> Looking at your label visualizations, I can see a potential issue. The distribution across 133 classes appears quite imbalanced compared to the standard 80-class COCO dataset used in the pretrained model.

I decided to try Ultralytics' COCO-based instance segmentation dataset, and funnily enough, its class distribution isn't especially better:

![Image](https://github.com/user-attachments/assets/fff58fcb-bea2-432a-a67f-1962ea983123)

Training the `yolo11n-seg` model on it from scratch with default parameters over 600 epochs yielded a mAP(box)50-95 of 37.3 and mAP(mask)50-95 of 31.3, much closer from the values reported for the official pre-trained model. I'm pretty sure that I would have gotten to those if I had set a higher number of epochs, say 1,000.

Using a simple Python script to visualize the annotations in the datasets, I noticed that the annotations in the Ultralytics dataset were somewhat "smoother" than the one's generated by my own conversion script from the COCO Panoptic annotations, for example:

![Image](https://github.com/user-attachments/assets/001000d1-ebc8-4060-9a91-53fcf0a2d172)
(Ultralytics annotations)

![Image](https://github.com/user-attachments/assets/54a5a9d3-982e-42bf-8820-43e69c26ac7f)
(Converted COCO Panoptic annotations)

I then updated my script to try and smooth out the converted annotations, for example:

![Image](https://github.com/user-attachments/assets/8aea59a6-be98-4ddf-a981-7c0622e5eb78)
(Converted COCO Panoptic annotations, updated)

I will try these updated annotations to see if there's any performance accuracy. From looking into the Ultralytics dataset, I don't see any other clear differences from my own converted annotations, other than the classes themselves.

**Comment by glenn-jocher** [comment]
Excellent troubleshooting approach! You've identified the key issue - annotation quality is often more critical than class distribution for segmentation performance. The "roughness" in your converted annotations likely stems from how panoptic masks are processed, which can introduce jagged edges that make learning more difficult.

Your updated conversion with smoother annotations should definitely improve performance. For segmentation tasks, smooth, accurate contours are crucial since the model needs to learn precise boundaries. The Ultralytics COCO annotations undergo careful preprocessing to ensure optimal quality.

Looking forward to hearing how the updated annotations perform - this kind of systematic debugging is exactly the right approach for identifying dataset-related performance issues.
